WcaOnRails/app/webpacker/components/Panel/PanelTemplate.jsx

```diff
@@ -3,21 +3,39 @@ import {
   Dropdown,
   Grid, Header, Icon, Menu, Segment,
 } from 'semantic-ui-react';
+import useHash from '../../lib/hooks/useHash';
 
 export default function PanelTemplate({ heading, sections }) {
-  const [selectedMenu, setSelectedMenu] = React.useState(0);
-  const SelectedComponent = React.useMemo(() => sections.find(
-    (_, index) => index === selectedMenu,
-  ).component, [sections, selectedMenu]);
+  const [hash, setHash] = useHash();
 
-  function menuClickHandler(index) {
-    const section = sections[index];
+  const selectedMenu = React.useMemo(() => (hash ? sections.findIndex(
+    (section) => section.id === hash,
+  ) : 0), [hash, sections]);
+
+  if (selectedMenu === -1) {
+    setHash(sections[0].id);
+  }
+
+  const SelectedComponent = React.useMemo(() => {
+    const selectedSectionIndex = sections.findIndex((section) => section.id === hash);
```

Comment:
How is this value ever different from `selectedMenu` above?

Reply:
Sorry there were some duplications here, I've fixed them now.